python/cudf/cudf/_lib/pylibcudf/io/types.pyx

```diff
@@ -108,3 +147,85 @@ cdef class SourceInfo:
                                                      c_buffer.shape[0]))
 
         self.c_obj = source_info(c_host_buffers)
+
+
+# Adapts a python io.IOBase object as a libcudf IO data_sink. This lets you
+# write from cudf to any python file-like object (File/BytesIO/SocketIO etc)
+cdef cppclass iobase_data_sink(data_sink):
```

Comment:
This part is mostly copied and adapted from ``make_sink_info`` in cudf._lib.io.utils

Reply:
It's a bit strange to define a `cppclass` in Cython. It would probably be better to switch it to a `cdef class` to avoid any oddness in semantics, but it's not a big deal.